docs/user/TroubleShooting.md

```diff
@@ -0,0 +1,123 @@
+
+For in-depth information and steps to replace
+a Master/Control Plane machine, please refer
+to this guide: https://docs.openshift.com/container-platform/4.5/backup_and_restore/replacing-unhealthy-etcd-member.html
+
+# Table of Contents
+<!-- toc -->
+- [Document Purpose](#document-purpose)
+- [Important Pod Logs](#important-pod-logs)
+  - [machine-api components](#machine-api-components)
+  - [cluster-machine-approver](#cluster-machine-approver)
+- [I created a Machine (or scaled up a MachineSet) but I didn't get a Node.](#i-created-a-machine-or-scaled-up-a-machineset-but-i-didnt-get-a-node)
+  - [Machine Status: Phase Provisioning](#machine-status-phase-provisioning)
+  - [Machine Status: Phase Provisioned](#machine-status-phase-provisioned)
+  - [Machine Status: Phase Failed](#machine-status-phase-failed)
+- [I deleted a Machine (or scaled down a MachineSet) but the Machine and/or Node did not go away](#i-deleted-a-machine-or-scaled-down-a-machineset-but-the-machine-andor-node-did-not-go-away)
+- [A Machine is listed as 'Failed'](#a-machine-is-listed-as-failed)
+<!-- /toc -->
+
+# Document Purpose
+This document outlines the steps to investigate the current status of an individual Machine or MachineSet that appears to not be creating Machines which results in additional Nodes joining the cluster.  Troubleshooting an unhealthy Node (a Machine that already joined the cluster as a Node) is outside the scope of this document.
+
+# Important Pod Logs
+## machine-api components
+Most everything that relates to the machine-api is viewable in the `openshift-machine-api` namespace.  You will want to familiarize yourself with the output of
+```sh
+oc get deployments -n openshift-machine-api
+oc get pods -n openshift-machine-api
+```
+
+The `machine-api-controllers-*` pod has several containers running: `machineset-controller`, `machine-controller`, `nodelink-controller`, and the `machine-healthcheck-controller`.
+
+To check the logs for a particular component, use
+```sh
+oc logs -n openshift-machine-api machine-api-controllers-<random suffix> -c <controller-name>
+```
+The random suffix is automatically generated by the `machine-api-controllers` deployment and is most easily found using the output of
+```sh
+oc get pods -n openshift-machine-api
+```
+
+## cluster-machine-approver
+CSRs that are automatically generated by kubelets on instances provisioned by the machine-api will automatically attempt to join the cluster by issuing a `CSR (certificate signing request)`.  Under normal circumstances, these CSRs should be approved automatically.  On rare occasions, you may encounter a bug where a CSR is stuck in pending state and the kubelet is unable to join the cluster successfully.
+
+To view the `cluster-machine-approver` logs, perform the following:
+
+```sh
+oc get pods -n openshift-cluster-machine-approver
+```
+
+Note the name of the pod `machine-approver-*` The suffix will be randomly generated by the pod's deployment controller.
```

Comment:
This isn't technically true (API server handles name generation), and it's mentioned a couple of times, do we need to mention this? Might be better to say that `a random suffix is appended to the pod name when the pod is created`?